src/UI/DatePicker.elm

```diff
@@ -0,0 +1,414 @@
+module UI.DatePicker exposing (DateEvent(..), DatePicker(..), Msg(..), datepicker, init, update)
+
+import Date exposing (Date, Interval(..), Month, Unit(..))
+import Element exposing (Element, alignLeft, alignRight, fill, fillPortion, height, maximum, minimum, padding, paddingXY, px, rgb, shrink, spacing, width)
+import Element.Border as Border
+import Element.Events
+import Element.Font as Font
+import Json.Decode
+import Time exposing (Month(..))
+import UI.Button as Button
+import UI.Icon as Icon
+import UI.Palette as Palette
+import UI.RenderConfig as RenderConfig exposing (RenderConfig)
+import UI.Size as Size
+import UI.Text as Text
+
+
+type alias Property msg =
+    { today : Date
+    , current : Date
+    , selected : Maybe Date
+    , toExternal : Msg -> msg
+    }
+
+
+type alias DayItem =
+    { date : Date
+    , isCurrentMonth : Bool
+    }
+
+
+type DatePicker msg
+    = DatePicker (Property msg)
+
+
+init : Time.Posix -> Maybe Date -> (Msg -> msg) -> DatePicker msg
+init today selected toExt =
+    let
+        todayDate =
+            Date.fromPosix Time.utc today
```

Comment:
We can't have the timezone hardcoded. We should receive a `Time.Zone` parameter instead of using `Time.utc`